src/internal/m365/collection/drive/collections.go

```diff
@@ -856,6 +863,22 @@ func (c *Collections) processItem(
 			PathPrefix(maps.Keys(topLevelPackages)).
 			Compare(collectionPath.String())
 
+		// This check is to ensure that if a folder was deleted and
+		// recreated multiple times between a backup, we only use the
+		// final one.
+		pid, found := seenFolders.IDOf(collectionPath.String())
+
+		recreated := found && pid != itemID
+		if recreated {
+			prevPath = nil
+			c.NumContainers--
+			c.NumItems--
```

Comment:
This feels a bit hazardous, but I'm not sure the alternatives are less clunky.  Thoughts?